packages/features/bookings/lib/handleNewBooking/getLocationValuesForDb.ts

```diff
@@ -42,9 +42,17 @@ const _getLocationValuesForDb = <
         credentials: firstDynamicGroupMember.credentials,
       });
 
-    firstDynamicGroupMemberDefaultLocationUrl =
-      firstDynamicGroupMemberMetadata?.defaultConferencingApp?.appLink ||
-      firstDynamicGroupMemberDelegationCredentialConferencingAppLocation;
+    const defaultConferencingApp = firstDynamicGroupMemberMetadata?.defaultConferencingApp;
+
+    const hasMemberSetConferencingPreference =
+      !!defaultConferencingApp?.appSlug || !!defaultConferencingApp?.appLink;
+
+    if (!hasMemberSetConferencingPreference) {
+      firstDynamicGroupMemberDefaultLocationUrl =
+        firstDynamicGroupMemberDelegationCredentialConferencingAppLocation ?? null;
+    } else {
+      firstDynamicGroupMemberDefaultLocationUrl = defaultConferencingApp?.appLink ?? null;
+    }
```

Comment:
this would be cleaner
```
firstDynamicGroupMemberDefaultLocationUrl = (hasMemberSetConferencingPreference ? defaultConferencingApp?.appLink : firstDynamicGroupMemberDelegationCredentialConferencingAppLocation) ?? null
```